include/datadog/baggage.h

```diff
@@ -0,0 +1,143 @@
+  ///
+  /// @param `baggage_map` The map containing key-value pairs to initialize the
+  /// Baggage.
+  /// @param `max_capacity` The maximum capacity for this Baggage instance.
+  Baggage(std::unordered_map<std::string, std::string>,
+          size_t max_capacity = default_max_capacity);
+
+  /// Initializes a Baggage instance using the provided initializer list of
+  /// key-value pairs. The maximum capacity can also be specified.
+  ///
+  /// @param `init_list` An initializer list containing key-value pairs.
+  /// @param `max_capacity` The maximum capacity for this Baggage instance.
+  Baggage(std::initializer_list<std::pair<const std::string, std::string>>,
+          size_t max_capacity = default_max_capacity);
+
+  /// Checks if the Baggage contains a specified key.
+  ///
+  /// @param `key` The key to check.
+  /// @return `true` if the key exists in the Baggage; otherwise, `false`.
+  bool contains(StringView key) const;
+
+  /// Retrieves the value associated with a specified key.
+  ///
+  /// @param `key` The key to retrieve the value for.
+  /// @return An `Optional<StringView>` containing the value if the key exists,
+  /// or an empty Optional if the key is not found.
+  Optional<StringView> get(StringView key) const;
+
+  /// Adds a key-value pair to the Baggage.
+  ///
+  /// This function will attempt to add the given key-value pair to the Baggage.
+  /// If the maximum capacity has been reached, the insertion will fail.
+  /// If a `key` already exists, its value will be overwritten with `value`.
+  ///
+  /// @param `key` The key to insert.
+  /// @param `value` The value to associate with the key.
+  /// @return `true` if the key-value pair was successfully added; `false` if
+  /// the maximum capacity was reached.
+  bool set(std::string key, std::string value);
+
+  /// Removes the key-value pair corresponding to the specified key.
+  ///
+  /// @param `key` The key to remove from the Baggage.
+  void remove(StringView key);
+
+  /// Removes all key-value pair.
+  void clear();
+
+  /// Retrieves the number of items stored.
+  size_t size() const;
+
+  /// Returns whether any items are stored.
+  bool empty() const;
+
+  /// Visits each key-value pair in the Baggage and invoke the provided
+  /// visitor function for each key-value pair in the Baggage.
+  ///
+  /// @param `visitor` A function object that will be called for each
+  /// key-value pair.
+  void visit(std::function<void(StringView, StringView)>&& visitor);
+
+  /// Injects the Baggage data into a `DictWriter` with the constraint that
+  /// the amount of bytes written does not exceed the specified maximum byte
+  /// limit.
+  ///
+  /// @param `writer` The DictWriter to inject the data into.
+  /// @param `max_bytes` The maximum number of bytes to write.
+  /// @return An `Expected<void>`, which may either succeed or contain an
+  /// error.
+  Expected<void> inject(DictWriter& writer, size_t max_bytes) const;
```

Comment:
What do you think of changing the `size_t max_bytes` argument into a `BaggageOptions` (you can choose the name) struct similar to `datadog::tracing::InjectionOptions`?

Reply:
Unless there is a need, I don't think so.

Reply:
I also don't see a direct need today, but I was thinking that it would be a better user that if we needed to add options then we could keep the function signature the same and just add to an options class. Anyways, this concern isn't a blocker

Reply:
Adressed in e3394c9f9e776b0f769c160914b30aeda6dea13a